promql/lex_test.go

```diff
@@ -429,6 +429,120 @@ var tests = []struct {
 		},
 		seriesDesc: true,
 	},
+	// Test subquery.
+	{
+		input: `test_name{on!~"bar"}[4m:4s]`,
+		expected: []item{
+			{itemIdentifier, 0, `test_name`},
+			{itemLeftBrace, 9, `{`},
+			{itemIdentifier, 10, `on`},
+			{itemNEQRegex, 12, `!~`},
+			{itemString, 14, `"bar"`},
+			{itemRightBrace, 19, `}`},
+			{itemLeftBracket, 20, `[`},
+			{itemDuration, 21, `4m`},
+			{itemColon, 23, `:`},
+			{itemDuration, 24, `4s`},
+			{itemRightBracket, 26, `]`},
+		},
+	},
+	{
+		input: `test:name{on!~"bar"}[4m:4s]`,
+		expected: []item{
+			{itemMetricIdentifier, 0, `test:name`},
+			{itemLeftBrace, 9, `{`},
+			{itemIdentifier, 10, `on`},
+			{itemNEQRegex, 12, `!~`},
+			{itemString, 14, `"bar"`},
+			{itemRightBrace, 19, `}`},
+			{itemLeftBracket, 20, `[`},
+			{itemDuration, 21, `4m`},
+			{itemColon, 23, `:`},
+			{itemDuration, 24, `4s`},
+			{itemRightBracket, 26, `]`},
+		},
+	}, {
+		input: `test:name{on!~"b:ar"}[4m:4s]`,
+		expected: []item{
+			{itemMetricIdentifier, 0, `test:name`},
+			{itemLeftBrace, 9, `{`},
+			{itemIdentifier, 10, `on`},
+			{itemNEQRegex, 12, `!~`},
+			{itemString, 14, `"b:ar"`},
+			{itemRightBrace, 20, `}`},
+			{itemLeftBracket, 21, `[`},
+			{itemDuration, 22, `4m`},
+			{itemColon, 24, `:`},
+			{itemDuration, 25, `4s`},
+			{itemRightBracket, 27, `]`},
+		},
+	}, {
+		input: `test:name{on!~"b:ar"}[4m:]`,
+		expected: []item{
+			{itemMetricIdentifier, 0, `test:name`},
+			{itemLeftBrace, 9, `{`},
+			{itemIdentifier, 10, `on`},
+			{itemNEQRegex, 12, `!~`},
+			{itemString, 14, `"b:ar"`},
+			{itemRightBrace, 20, `}`},
+			{itemLeftBracket, 21, `[`},
+			{itemDuration, 22, `4m`},
+			{itemColon, 24, `:`},
+			{itemRightBracket, 25, `]`},
+		},
+	}, { // Nested Subquery.
+		input: `min_over_time(sum(rate(testname{foo="bar"}[5m])[4m:4s])[5m:5s])`,
```

Comment:
The nested subquery isn't needed here, do you have a better example?

Reply:
Are we not planning to support nested subquery? (I haven't used promql much, so I am unaware of use cases)